### PR TITLE
[MODULAR] Fixes Quickspawn/SimpleTF/genetics eating mutcols

### PR DESCRIPTION
**Pull request by CliffracerX**
## About The Pull Request

Fixes an underlying error in Skyrat's DNA code, initially thought to be a problem with safe_transfer_prefs_to; it's been pulling mutcolors from your ident DNA for ages and somehow only broke in a few niche cases.  This fixes that.

## How This Contributes To The Skyrat Roleplay Experience

Makes thunderdome testing, event setup, and likely genetics alterations to feature/ident DNA stop turning people funny colors

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>

https://github.com/Skyrat-SS13/Skyrat-tg/assets/2958111/f957d4cc-1de1-42eb-a876-280b6835c634

</details>

## Changelog

:cl:
fix: mutantcolors are being pulled from your features dna again, no more pink feathers!
/:cl:

### CODE REVIEW

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -542,7 +542,8 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 /// Sanitizes the preferences, applies the randomization prefs, and then applies the preference to the human mob.
 /datum/preferences/proc/safe_transfer_prefs_to(mob/living/carbon/human/character, icon_updates = TRUE, is_antag = FALSE)
 	apply_character_randomization_prefs(is_antag)
-	apply_prefs_to(character, icon_updates)
+	spawn(1) /// SKYRAT EDIT: delaying this slightly to resolve some wacky timing bugs that eat charprefs
+		apply_prefs_to(character, icon_updates)
```
Comment (GoldenAlpharex) [inline code/modules/client/preferences.dm]: Would you mind giving something like `IMMEDIATE_INVOKE_ASYNC()` (or whatever it's actually spelled like) a try? I'd rather avoid using `spawn()` if we can, it's something we try and avoid as much as possible in the codebase, for debugging reasons.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

File: code/modules/client/preferences.dm (hunk 1 above)
Comment (CliffracerX) [reply code/modules/client/preferences.dm]: No prob, I didn't know spawn() had debug issues!

Comment (CliffracerX) [reply code/modules/client/preferences.dm]: ADDENDUM: I tried it with INVOKE_ASYNC and it broke in *new* ways.

Comment (CliffracerX) [reply code/modules/client/preferences.dm]: Put in a little wrapper function that sleeps to restore the expected behaviour while still using INVOKE_ASYNC.  It's a wee bit ugly but it works as expected at least.

**Review by lessthnthree — APPROVED** [review]

**Review by GoldenAlpharex — DISMISSED** [review]
I still feel like this is a hacky solution, and that it's sweeping the actual issue under the rug rather than searching to fix it.

**Review by lessthnthree — APPROVED** [review]

**Pull request merged**